src/lib/Instance.ts

```diff
@@ -0,0 +1,24 @@
+export default class Instance {
+  private _onChanged: ((...args: any[]) => void)[];
+  onChanged: {
+    connect: (callback: (...args: any[]) => void) => () => void;
+    fire: (...args: any[]) => void;
+  };
+
+  constructor() {
+    this._onChanged = [];
+    this.onChanged = {
+      connect: (callback: (...args: any[]) => void) => {
+        const length = this._onChanged.length;
+        this._onChanged.push(callback);
+
+        return () => {
+          this._onChanged.splice(length, 1);
+        };
+      },
+      fire: (...args: any[]) => {
+        this._onChanged.forEach((callback) => callback(args));
```

Comment:
```suggestion
        this._onChanged.forEach((callback) => setTimeout(()=>callback(args),0));
```
Run on next event loop in parallel, prevent any of them throwing an error cause the other ones to not run (without silencing error altogether, etc...)